crates/sui-core/src/authority_aggregator.rs

```diff
@@ -2256,3 +2262,83 @@ pub async fn reconfig_from_genesis(
     // Now transit from latest_epoch - 1 to latest_epoch
     aggregator.recreate_with_net_addresses(latest_committee, &network_config)
 }
+
+pub struct AuthorityAggregatorBuilder<'a> {
+    network_config: Option<&'a NetworkConfig>,
+    genesis: Option<&'a Genesis>,
+    committee_store: Option<Arc<CommitteeStore>>,
+    registry: Option<Arc<Registry>>,
+}
+
+impl<'a> AuthorityAggregatorBuilder<'a> {
+    pub fn from_network_config(config: &'a NetworkConfig) -> Self {
+        Self {
+            network_config: Some(config),
+            genesis: None,
+            committee_store: None,
+            registry: None,
+        }
+    }
+
+    pub fn from_genesis(genesis: &'a Genesis) -> Self {
+        Self {
+            network_config: None,
+            genesis: Some(genesis),
+            committee_store: None,
+            registry: None,
+        }
+    }
+
+    pub fn with_committee_store(mut self, committee_store: Arc<CommitteeStore>) -> Self {
+        self.committee_store = Some(committee_store);
+        self
+    }
+
+    pub fn with_registry(mut self, registry: Arc<Registry>) -> Self {
+        self.registry = Some(registry);
+        self
+    }
+
+    pub fn build(
+        self,
+    ) -> anyhow::Result<(
+        AuthorityAggregator<NetworkAuthorityClient>,
+        BTreeMap<AuthorityPublicKeyBytes, NetworkAuthorityClient>,
+    )> {
+        let validator_info = if let Some(network_config) = self.network_config {
+            network_config.validator_set()
+        } else if let Some(genesis) = self.genesis {
+            genesis.validator_set()
```

Comment:
Why do we need to pass genesis blob if all we need is the validator set from it?

Reply:
this is a fair question. All existing callsites takes `NetworkConfig` (or `GatewayConfig` which I'm partially eliminating), so it's naturally to take `Genesis` v.s. `&[ValidatorInfo]`

I don't have preferences to one way or the other. I can change to using validators array if you feel strongly